ui/raidboss/data/07-dt/dungeon/meso-terminal.ts

```diff
@@ -0,0 +1,405 @@
+      // AB1D is small
+      id: 'Meso Terminal Chirurgeon Sterile Sphere Collect',
+      type: 'StartsUsingExtra',
+      netRegex: { id: 'AB1D', capture: true },
+      run: (data, matches) => {
+        const location = {
+          x: Math.round(parseFloat(matches.x)),
+          y: Math.round(parseFloat(matches.y)),
+        };
+        data.chirurgeonSpots.push(location);
+      },
+    },
+    {
+      id: 'Meso Terminal Chirurgeon Sterile Sphere Call',
+      type: 'StartsUsing',
+      netRegex: { id: 'AB1D', source: 'Chirurgeon General', capture: false },
+      delaySeconds: 1,
+      suppressSeconds: 1,
+      alertText: (data, _matches, output) => {
+        if (!data.chirurgeonSpots[0] || !data.chirurgeonSpots[1])
+          return;
+        const spot1 = data.chirurgeonSpots[0];
+        const spot2 = data.chirurgeonSpots[1];
+        // Both spheres east or both west.
+        // X value under 270 means east safe.
+        if (spot1.x === spot2.x) {
+          if (spot1.x < 270)
+            return output.goEast!();
+          return output.goWest!();
+        }
+        // Both spheres north or south.
+        // Y values increase from north to south.
+        // Y value over 12 means north safe.
+        if (spot1.y === spot2.y) {
+          if (spot1.y > 12)
+            return output.goNorth!();
+          return output.goSouth!();
+        }
+        return output.unknown!();
+      },
+      run: (data) => data.chirurgeonSpots = [],
+      outputStrings: {
+        goNorth: Outputs.north,
+        goEast: Outputs.east,
+        goSouth: Outputs.south,
+        goWest: Outputs.west,
+        unknown: Outputs.unknown,
+      },
+    },
+    {
+      id: 'Meso Terminal Chirurgeon Concentrated Dose',
+      type: 'StartsUsing',
+      netRegex: { id: 'AB17', source: 'Chirurgeon General', capture: true },
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'Meso Terminal Executioners Shackles Of Fate Tracking',
+      type: 'StartsUsing',
+      netRegex: { id: 'AA3D', capture: true },
+      condition: Conditions.targetIsYou(),
+      run: (data, matches) => data.playerExecutionerId = matches.sourceId,
+    },
+    {
+      id: 'Meso Terminal Executioners Head-Splitting Roar',
+      type: 'StartsUsing',
+      netRegex: { id: 'AA3B', source: 'Hooded Headsman', capture: false },
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Meso Terminal Executioners Chopping Block',
+      type: 'StartsUsing',
+      netRegex: { id: 'AA4B', capture: true },
+      condition: (data, matches) => data.playerExecutionerId === matches.sourceId,
+      response: Responses.getOut(),
+    },
+    {
+      id: 'Meso Terminal Executioners Execution Wheel',
+      type: 'StartsUsing',
+      netRegex: { id: 'AA4C', capture: true },
+      condition: (data, matches) => data.playerExecutionerId === matches.sourceId,
+      response: Responses.getIn(),
+    },
+    {
+      id: 'Meso Terminal Executioners Flaying Flail',
```

Comment:
This is probably fine for now. Without overlayplugin to tell where the player is in relation to which adds are doing the flail, this could call for players that are not receiving the flails.

Possible to also do this loosely with a filter for data.role and the target location of the flail since the role locations are static. I think maybe DPS get the flails at one time and tank/healer get the flails at a different time?

Reply:
As far as I know, all flails are aligned if there are no deaths, or at least that's what my recordings seem to say. I'd rather not get too into the weeds with trying to support misalignments due to death, since collecting data for it would be kind of miserable.

Reply:
I think it's the odd flails that target everyone, evens target only dps? The 2nd flail the supports get a tb/doom to cleanse while the dps get flails.

Reply:
Oh, I see what you mean. Filtering by roles might still be complex, but what do you think of adding the same filtering-by-chain-target condition as we use for Chopping Block and Execution Wheel, or is that what you meant?

Reply:
It looks like the `playerExecutionerId` sourceId is always different from the Flaying Flail (AA48) sourceId. But if source is used for playerExecutionerId it matches, at least in English locale.

Reply:
The player-tethered Executioners use `AA47` as a pre-cast before the flails appear and the `AA48` cast begins. What about something like this?

```javascript
    {
      id: 'Meso Terminal Executioners Flaying Flail',
      type: 'Ability,
      netRegex: { id: 'AA47', capture: true },
      condition: (data, matches) => data.playerExecutionerId === matches.sourceId,
      // etc etc
    },
```
And then we could use a `delaySeconds` value if needed to make it comfortable. At least in my logs, the sources are consistent for this usage.

Reply:
Ah, the sourceId is also different on that AA48 cast than the entity that is casting the shackles. There are 3 entities that each cast the AA48 per shackle.